src/userdev/java/net/minecraftforge/gradle/userdev/UserDevPlugin.java

```diff
@@ -154,18 +172,21 @@ public RenameJarInPlace create(String jarName) {
                 minecraft.getDependencies().add(ext);
             }
 
-            deps = deobf.getDependencies();
-            for (Dependency dep : deps.stream().collect(Collectors.toList())) {
-                if (!(dep instanceof ExternalModuleDependency)) //TODO: File deps as well.
-                    throw new IllegalArgumentException("deobf dependency must be a maven dependency. File deps are on the TODO");
-                deps.remove(dep);
+            DependencySet legacyDeps = deobfConfiguration.getDependencies();
```

Comment:
If this is a legacy list, then add a //TODO: Remove FG3.1 or something so we can remove it. I'd rather not have a long lasting list of depreciated code.